### ISSUE TITLE
Unable to start service

### ISSUE THREAD

**Issue by addisjeams** [gpustack/gpustack#633]
PS C:\Users\win10go> gpustack version
0.3.2 (43077e2)

I tried installing gpustack on two computers, version 0.3.2 (43077e2).
﻿
Two issues have arisen
﻿
Problem 1: After server A is installed, it does not recognize its own worker. If you want to add this machine as a worker, please write it clearly on the instruction page.
In addition, another computer B, which serves as a worker, cannot connect to server A. By entering an IP address, it can see web pages at times, but at other times it cannot. The ping machine is capable of pinging.
I entered the following two sentences on worker computer B, but neither of them can join the cluster
﻿
Invoke-Expression "& { $((Invoke-WebRequest -Uri "https://get.gpustack.ai" -UseBasicParsing).Content) } --server-url http://192.168.0.199 --token f2eb0529df2ca1c587aa6ba442d340b7"

Invoke-Expression "& { $((Invoke-WebRequest -Uri 'https://get.gpustack.ai' -UseBasicParsing).Content) } --server-url http://192.168.0.199 --token f2eb0529df2ca1c587aa6ba442d340b7"

In summary, the formation of the cluster was not completed.
﻿
-------------------------------------------------------------------------
Problem 2: After server A restarts, I can no longer connect to the service, appearing as unable to open web pages, as if the web service is not even turned on.
I entered gpustack start, and the system reported that it has already started
I manually closed python.exe that was occupying port 80, but there was no way.
﻿
﻿---------------------------------------------------------------

Both computers run on Windows 10 and are equipped with Nvidia graphics cards

**Comment by gitlawr** [comment]
Can you provide GPUStack logs following https://docs.gpustack.ai/latest/troubleshooting/#windows

**Comment by addisjeams** [comment]
[gpustack.log](https://github.com/user-attachments/files/17942229/gpustack.log)

**Comment by gitlawr** [comment]
There are two issues from the logs.
1. It cannot find a valid IP to use. Can you show `ipconfig` of the node. Alternatively, it can be specified using `--worker-ip` as a workaround if it's considered to be a false alarm.
2. It fails to download prebuilt tools from github. We provide an alternative source in the cloud, which can be specified by `--tools-download-base-url https://gpustack-1303613262.cos.ap-guangzhou.myqcloud.com`. I'm not sure if that works in your environment.

**Comment by addisjeams** [comment]
I will try it out when I have time and continue to provide feedback here.

Additionally, I have another question. If possible, could you guide me on how to use the local gguf model? I have downloaded them all from the internet. We need a private deployment, completely offline, which is a very strong and necessary requirement. We can download it initially, but we won't be able to connect to the internet later on.

**Comment by gitlawr** [comment]
It is available in the upcoming v0.4.0: https://github.com/gpustack/gpustack/issues/418
If you are interested here's how the experience looks like(main & 0.4.0rc1 right now): https://github.com/gpustack/gpustack/blob/main/docs/user-guide/model-management.md#deploying-a-local-path-model

**Comment by pengjiang80** [comment]
Close as there is no update for more than 3 months. Feel free to re-open it as required.

**Comment by sdfflq** [comment]
logs：
2025-03-12T15:34:16+08:00 - gpustack.worker.worker_manager - WARNING - RPC server for GPU 0 is not running, pid 1988840, restarting.
2025-03-12T15:34:16+08:00 - gpustack.worker.worker_manager - INFO - Started RPC server for GPU 0 on port 50546, pid 1989797

**Comment by gitlawr** [comment]
@sdfflq Can you check logs in `/var/lib/gpustack/log/rpc_server/gpu-0.log`? Given the differences between GPUStack versions, your case is likely not the same with the original issue. I recommend opening a new issue if you need further discussion.

**Comment by sdfflq** [comment]
> [@sdfflq](https://github.com/sdfflq) Can you check logs in ? Given the differences between GPUStack versions, your case is likely not the same with the original issue. I recommend opening a new issue if you need further discussion.`/var/lib/gpustack/log/rpc_server/gpu-0.log`


2025-03-12T16:02:41+08:00 - gpustack.worker.worker_manager - INFO - Started RPC server for GPU 0 on port 50378, pid 2067526
2025-03-12T16:03:01+08:00 - gpustack.worker.worker_manager - WARNING - RPC server for GPU 0 is not running, pid 2067526, restarting.
2025-03-12T16:03:01+08:00 - gpustack.worker.worker_manager - INFO - Started RPC server for GPU 0 on port 50408, pid 2068501
2025-03-12T16:03:21+08:00 - gpustack.worker.worker_manager - WARNING - RPC server for GPU 0 is not running, pid 2068501, restarting.
2025-03-12T16:03:21+08:00 - gpustack.worker.worker_manager - INFO - Started RPC server for GPU 0 on port 50958, pid 2069469
2025-03-12T16:03:41+08:00 - gpustack.worker.worker_manager - WARNING - RPC server for GPU 0 is not running, pid 2069469, restarting.
2025-03-12T16:03:41+08:00 - gpustack.worker.worker_manager - INFO - Started RPC server for GPU 0 on port 50194, pid 2070457
2025-03-12T16:04:01+08:00 - gpustack.worker.worker_manager - WARNING - RPC server for GPU 0 is not running, pid 2070457, restarting.
2025-03-12T16:04:01+08:00 - gpustack.worker.worker_manager - INFO - Started RPC server for GPU 0 on port 50994, pid 2071438

**Comment by sdfflq** [comment]
> [@sdfflq](https://github.com/sdfflq) Can you check logs in `/var/lib/gpustack/log/rpc_server/gpu-0.log`? Given the differences between GPUStack versions, your case is likely not the same with the original issue. I recommend opening a new issue if you need further discussion.

![Image](https://github.com/user-attachments/assets/0901d6a2-695b-4735-9473-32077c4624b9)

**Comment by gitlawr** [comment]
We would like to check `/var/lib/gpustack/log/rpc_server/gpu-0.log` instead of `/var/log/gpustack.log`. Is that file missing or empty?

**Comment by gitlawr** [comment]
I mean `/var/lib/gpustack/xxx`, not `/var/log/xxx`

**Comment by sdfflq** [comment]
There is no such file.

**Comment by sdfflq** [comment]
> I mean `/var/lib/gpustack/xxx`, not `/var/log/xxx`

[

[gpu-0.log](https://github.com/user-attachments/files/19206151/gpu-0.log)

](url)

**Comment by gitlawr** [comment]
Can you run `nvcc --version` and `nvidia-smi` to check CUDA, and provide info of your setup

**Comment by sdfflq** [comment]
'nvcc' not found

**Comment by sdfflq** [comment]
![Image](https://github.com/user-attachments/assets/8dae799b-f66d-4756-b6f5-0bbac17a57d4)

**Comment by gitlawr** [comment]
@sdfflq Please follow instructions in https://github.com/gpustack/gpustack/wiki/FAQ#how-can-i-resolve-the-error-so-cannot-open-shared-object-file-no-such-file-or-directory
